models/classes/user/UserSettingsFormFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\user;
+
+use tao_helpers_form_FormContainer;
+use tao_actions_form_UserSettings;
+use tao_helpers_form_Form;
+
+class UserSettingsFormFactory
+{
+    /** @var UserSettingsInterface */
+    private $userSettings;
+
+    /** @var string */
+    private $defaultLanguage;
+
+    public function __construct(UserSettingsInterface $userSettings, string $defaultLanguage)
+    {
+        $this->userSettings = $userSettings;
+        $this->defaultLanguage = trim($defaultLanguage);
+    }
+
+    public function getForm(bool $csrf = true): tao_helpers_form_Form
```

Comment:
Some suggestions:

- This is a factory, so it `creates`. Maybe `create()` better represents the intent of this?
- Use extensible parameters for public interface when possible. I suggest at least an array `array $params` 
- Having bool parameter for public/protected contract is a code smell and we should avoid. More details here https://martinfowler.com/bliki/FlagArgument.html

```suggestion
    public function create(array $params): tao_helpers_form_Form
```

And inside you can get `$params['useCSRFProtection']`, for example.

Reply:
Changed: b41d8e620299f0c5d980c37bbbf1b92a9c791da8